compiler-core/src/language_server/configuration.rs

```diff
@@ -0,0 +1,33 @@
+use serde::Deserialize;
+use std::sync::{Arc, RwLock};
+
+pub type SharedConfig = Arc<RwLock<Configuration>>;
+
+#[derive(Debug, Default, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+    #[serde(default = "InlayHintsConfig::default")]
+    pub inlay_hints: InlayHintsConfig,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InlayHintsConfig {
+    /// Whether to show type inlay hints of multiline pipelines
+    #[serde(default = "InlayHintsConfig::default_pipelines")]
+    pub pipelines: bool,
+}
+
+impl Default for InlayHintsConfig {
+    fn default() -> Self {
+        Self {
+            pipelines: Self::default_pipelines(),
+        }
+    }
+}
+
+impl InlayHintsConfig {
+    fn default_pipelines() -> bool {
+        false
```

Comment:
Note: this means that until the client adds this option to its config, the option won't be available. Are we sure we want this to be false by default?

Reply:
Tbh this is such a revolutionary addition that I think it should be on by default

Reply:
Does it differentiate between having pipes on separate lines or on one line? I think personally I would like to enable this feature for separate lines, but not for one line pipes.

Reply:
Yes it does (you can double check it by looking at the tests)
It also avoids showing hints of values that are int, float, string or bitstring literals

Reply:
It must be off by default, and it should be possible to turn it on momentarily.

Reply:
Hi Louis, to turn on the config option, the option must be defined in the client. In order to do that, you have to run the client PR in debug mode https://github.com/gleam-lang/vscode-gleam/pull/82

https://github.com/gleam-lang/gleam/assets/26420752/31c4cfc4-88ea-4e1d-b081-4c4ec9407d15

If there is a way to define the client option directly in gleam, without having the create a PR for every client extension, that would be great, but I'm not aware of it

Reply:
I think it's only VSCode that uses an extension, so it's not a big problem.

It looks like VSCode is reloading the entire language server there, which isn't applicable to any other editor. Some solution within the protocol would be good.

Reply:
> It must be off by default, and it should be possible to turn it on momentarily. How does the programmer do this? I couldn't see a way

In helix I use the following so I just press SPACE + i to toggle the option:

```toml
[keys.normal.space]
i = ":toggle lsp.display-inlay-hints"
```

Haven't checked if it works with this PR though

Reply:
I now see that this is an option specific to helix: 

*You may also have to activate them in the LSP config for them to appear, not just in Helix.*

https://github.com/helix-editor/helix/blob/fc97ecc3e3186b9dfe958869178bdb6b8cd7d8df/book/src/editor.md?plain=1#L126C7-L127C1